### PR TITLE
add static loading info for gdal data sets

### PR DESCRIPTION
**Pull request by michaelmattig**

### CODE REVIEW

**Comment by michaelmattig** [comment]
[rasterisierte Länder-Dataset hinzufügen](https://trello.com/c/eVrztPAO/338-rasterisierte-l%C3%A4nder-dataset-hinzuf%C3%BCgen)

File: operators/src/source/gdal_source.rs [hunk 1]
```diff
@@ -96,6 +96,31 @@ pub enum FileNotFoundHandling {
...
+pub struct GdalMetaDataStatic {
+    pub params: GdalDataSetParameters,
+    pub result_descriptor: RasterResultDescriptor,
+}
+
+impl MetaData<GdalLoadingInfo, RasterResultDescriptor> for GdalMetaDataStatic {
+    fn loading_info(&self, _query: QueryRectangle) -> Result<GdalLoadingInfo> {
+        Ok(GdalLoadingInfo {
+            info: vec![GdalLoadingInfoPart {
+                time: TimeInterval::default(),
```
Comment (ChristianBeilschmidt) [inline operators/src/source/gdal_source.rs:109]: Shouldn't we derive this from the supplied data?

File: services/src/datasets/in_memory.rs [hunk 2]
```diff
@@ -97,7 +99,14 @@ impl HashMapStorable for StaticMetaData<MockDataSetDataSourceLoadingInfo, Vector
...
     fn store(&self, id: InternalDataSetId, db: &mut HashMapDataSetDb) -> TypedResultDescriptor {
-        db.gdal_data_sets.insert(id, self.clone());
+        db.gdal_data_sets.insert(id, Box::new(self.clone()));
+        self.result_descriptor.clone().into()
+    }
+}
+
+impl HashMapStorable for GdalMetaDataStatic {
+    fn store(&self, id: InternalDataSetId, db: &mut HashMapDataSetDb) -> TypedResultDescriptor {
+        db.gdal_data_sets.insert(id, Box::new(self.clone()));
```
Comment (ChristianBeilschmidt) [inline services/src/datasets/in_memory.rs:109]: Can't you implement it for everything with certain trait bounds? Looks the same for me.

File: services/src/datasets/in_memory.rs [hunk 3]
```diff
@@ -226,18 +235,17 @@ impl MetaDataProvider<GdalLoadingInfo, RasterResultDescriptor> for HashMapDataSe
...
+            .gdal_data_sets
+            .get(&data_set.internal().ok_or(
+                geoengine_operators::error::Error::DataSetMetaData {
+                    source: Box::new(error::Error::DataSetIdTypeMissMatch),
+                },
+            )?)
+            .ok_or(geoengine_operators::error::Error::DataSetMetaData {
+                source: Box::new(error::Error::UnknownDataSetId),
+            })?
+            .clone())
```
Comment (ChristianBeilschmidt) [inline services/src/datasets/in_memory.rs:248]: This is super hard to read, maybe you can just use a match block?

**Review by ChristianBeilschmidt — CHANGES_REQUESTED** [review]
just some minor stuff

File: operators/src/source/gdal_source.rs (hunk 1 above)
Comment (michaelmattig) [reply operators/src/source/gdal_source.rs:109]: 1b1d88b8c3cd15e4c0b5d1f3663b94c92124cfe1

File: services/src/datasets/in_memory.rs (hunk 2 above)
Comment (michaelmattig) [reply services/src/datasets/in_memory.rs:109]: I tried
```
impl<G> HashMapStorable for G
where
    G: MetaData<GdalLoadingInfo, RasterResultDescriptor>,
{
    fn store(&self, id: InternalDataSetId, db: &mut HashMapDataSetDb) -> TypedResultDescriptor {
        db.gdal_data_sets.insert(id, Box::new(self.clone()));
        self.result_descriptor.clone().into()
    }
}
```

but Rust wouldn't let me:
> error[E0119]: conflicting implementations of trait `datasets::in_memory::HashMapStorable` for type `geoengine_operators::engine::StaticMetaData<geoengine_operators::source::OgrSourceDataset, geoengine_operators::engine::VectorResultDescriptor>`:
   --> services/src/datasets/in_memory.rs:93:1
    |
86  |   impl HashMapStorable for StaticMetaData<OgrSourceDataset, VectorResultDescriptor> {
    |   --------------------------------------------------------------------------------- first implementation here
...
93  | / impl<G> HashMapStorable for G
94  | | where
95  | |     G: MetaData<GdalLoadingInfo, RasterResultDescriptor>,
96  | | {
...   |
100 | |     }
101 | | }
    | |_^ conflicting implementation for `geoengine_operators::engine::StaticMetaData<geoengine_operators::source::OgrSourceDataset, geoengine_operators::engine::VectorResultDescriptor>`
    |
    = note: upstream crates may add a new impl of trait `geoengine_operators::engine::MetaData<geoengine_operators::source::GdalLoadingInfo, geoengine_operators::engine::RasterResultDescriptor>` for type `geoengine_operators::engine::StaticMetaData<geoengine_operators::source::OgrSourceDataset, geoengine_operators::engine::VectorResultDescriptor>` in future versions

File: services/src/datasets/in_memory.rs (hunk 3 above)
Comment (michaelmattig) [reply services/src/datasets/in_memory.rs:248]: on what should I match? I can split it up a bit if that helps:

```
        let id = data_set
            .internal()
            .ok_or(geoengine_operators::error::Error::DataSetMetaData {
                source: Box::new(error::Error::DataSetIdTypeMissMatch),
            })?;

        Ok(self
            .gdal_data_sets
            .get(&id)
            .ok_or(geoengine_operators::error::Error::DataSetMetaData {
                source: Box::new(error::Error::UnknownDataSetId),
            })?
            .clone())
```

Comment (michaelmattig) [reply services/src/datasets/in_memory.rs]: d1fc0c0f3586c3aee2d72918ee459b3fd25759ec

**Review by ChristianBeilschmidt — APPROVED** [review]

**Pull request merged**